plugin/completion/flags_manager.py

```diff
@@ -298,10 +299,9 @@ def write_flags_to_file(self, new_flags, file_path, strategy):
             if flag_strategy == "keep_old":
                 return
             if flag_strategy == "merge":
-                # union of two flags sets
-                curr_flags = set(self.flags_from_clang_file(
-                    File(file_path)))
-                new_flags = new_flags.union(curr_flags)
+                # union of two flag arrays
+                curr_flags = self.flags_from_clang_file(File(file_path))
+                new_flags = curr_flags + list(set(new_flags) - set(curr_flags))
```

Comment:
This code will put all new flags at the end of the list but I don't see it being either correct or incorrect really. As this code is only used when using cmake and generating the project, in practice the damage was already done in flags_from_database() when reading flags from cmake generated file (flags from different compilation units were merged together and the order is unspecified when doing that).